### PR TITLE
OSX compatibility

### PR DESCRIPTION
**Pull request by Rudloff**
Hello,

Here is a patch that allows nethogs to build on OSX.
It is not perfect, mainly because the /proc folder does not exist on OSX so I did not find any alternative to use in `reread_mapping()`, but everything else seems to work correctly.

### CODE REVIEW

File: conninode.cpp [hunk 1]
```diff
@@ -179,12 +185,17 @@ void refreshconninode ()
 	//delete conninode;
 	//conninode = new HashTable (256);
 
+	#if defined(__APPLE__)
+		addprocinfo("net.inet.tcp.pcblist");
+	#else
```
Comment (raboof) [inline conninode.cpp]: I suppose we should indent the non-apple code in the 'else' here

File: inode2prog.cpp [hunk 2]
```diff
@@ -225,7 +225,9 @@ struct prg_node * findPID (unsigned long inode)
 		return node;
 	}
 
-	reread_mapping();
+	#ifndef __APPLE__
+		reread_mapping();
```
Comment (raboof) [inline inode2prog.cpp]: If I understand correctly, this means on apple you only see connections, not processes. Is that accurate?

I suppose that's better than nothing, but of course nethogs' selling point is that you see the processes :)

**Comment by raboof** [comment]
Neat. Perhaps we should add a prominent note to the README that while it kind of works on OSX, it's more limited than on Linux?

Would be great of course if we could find a way to implement reread_mapping on OSX, too - but that could be a future PR :).

File: nethogs.cpp [hunk 3]
```diff
@@ -128,10 +128,10 @@ int process_tcp (u_char * userdata, const dp_header * header, const u_char * m_p
 	switch (args->sa_family)
 	{
 		case (AF_INET):
-			packet = new Packet (args->ip_src, ntohs(tcp->source), args->ip_dst, ntohs(tcp->dest), header->len, header->ts);
+			packet = new Packet (args->ip_src, ntohs(tcp->th_sport), args->ip_dst, ntohs(tcp->th_dport), header->len, header->ts);
```
Comment (Rudloff) [inline nethogs.cpp]: Can you confirm this would still work on Linux?
I tried it on Ubuntu and it seemed to work.

File: inode2prog.cpp (hunk 2 above)
Comment (Rudloff) [inline inode2prog.cpp]: Yes, you're right. But I think nethogs is still useful like this.
(And we might find a way to rewrite this function for OSX later.)

File: nethogs.cpp (hunk 3 above)
Comment (Rudloff) [inline nethogs.cpp]: Hum, it fails the Travis test, I will add a condition.

**Comment by Rudloff** [comment]
Would you like me to edit the README or do you want to do it yourself?

File: nethogs.cpp (hunk 3 above)
Comment (raboof) [inline nethogs.cpp]: Jep, those are synonymous in netinet/tcp.h (union with fields of the same type).

Presumably for compatibility with linux/tcp.h, but I have no problems with depending on netinet specifics.

**Comment by raboof** [comment]
Go for it!

**Comment by Rudloff** [comment]
Is it OK like this?

**Comment by raboof** [comment]
Yes, thank you very much for this contribution!

**Pull request merged**